domain/secret/state/state.go

```diff
@@ -191,14 +347,52 @@ ON CONFLICT(id) DO UPDATE SET
 	return nil
 }
 
-func (st State) upsertSecretModelOwner(ctx context.Context, tx *sqlair.TX, owner secretOwner) error {
+func (st State) upsertSecretModelOwner(ctx context.Context, tx *sqlair.TX, owner secretModelOwner) error {
 	insertQuery := `
 INSERT INTO secret_model_owner (secret_id, label)
-VALUES      ($secretOwner.*)
+VALUES      ($secretModelOwner.*)
 ON CONFLICT(secret_id) DO UPDATE SET label=excluded.label
 `
 
-	insertStmt, err := st.Prepare(insertQuery, secretOwner{})
+	insertStmt, err := st.Prepare(insertQuery, secretModelOwner{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = tx.Query(ctx, insertStmt, owner).Run()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func (st State) upsertSecretApplicationOwner(ctx context.Context, tx *sqlair.TX, owner secretApplicationOwner) error {
+	insertQuery := `
+INSERT INTO secret_application_owner (secret_id, application_uuid, label)
+VALUES      ($secretApplicationOwner.*)
+ON CONFLICT(secret_id, application_uuid) DO UPDATE SET label=excluded.label
```

Comment:
For update, is it the .UpdateSecretApplicationOwner responsible for getting the existing label and fill into the `secretApplicationOwner`?